wallet/wallet.c

```diff
@@ -2682,6 +2682,17 @@ void wallet_payment_delete(struct wallet *wallet,
 	db_exec_prepared_v2(take(stmt));
 }
 
+void wallet_payment_delete_by_hash(struct wallet *wallet,
+				   const struct sha256 *payment_hash)
+{
+	struct db_stmt *stmt;
+	stmt = db_prepare_v2(
+		wallet->db, SQL("DELETE FROM payments WHERE payment_hash = ?"));
```

Comment:
Looks correct but @cdecker should check as well, I am no longer familiar with how we interact with the db.

Reply:
This should work just fine. The SQL subsystem just takes a valid query and rewrites it for various dialects. That works well for this type of simple queries, but may be a bit more complex for things like upserts or last inserted IDs, which we use in some places.